### PR TITLE
cni: use hash to generate host-side veth name

### PR DESCRIPTION
**Pull request by nirmoy**
if CNI_CONTAINERID has 1st 5 char same then cni fails
chapman:/home/nirmoy/go/src/github.com/cilium/cilium # CNI_COMMAND=ADD CNI_CONTAINERID=1234567891 CNI_NETNS=/var/run/netns/1234567891 CNI_IFNAME=eth12 CNI_PATH=/opt/cni/bin ./plugins/cilium-cni/cilium-cni < ./plugins/cilium-cni/05-cilium-cni.conf  2>/dev/null >/dev/null
chapman:/home/nirmoy/go/src/github.com/cilium/cilium # CNI_COMMAND=ADD CNI_CONTAINERID=1234567892 CNI_NETNS=/var/run/netns/1234567892 CNI_IFNAME=eth12 CNI_PATH=/opt/cni/bin ./plugins/cilium-cni/cilium-cni < ./plugins/cilium-cni/05-cilium-cni.conf
{
    "code": 100,
    "msg": "unable to create veth pair: file exists"

Signed-off-by: Nirmoy Das <ndas@suse.de>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5620)

### CODE REVIEW

File: pkg/endpoint/connector/add.go [hunk 1]
```diff
@@ -37,7 +38,8 @@ const (
 
 // Endpoint2IfName returns the host interface name for the given endpointID.
 func Endpoint2IfName(endpointID string) string {
-	return hostInterfacePrefix + truncateString(endpointID, 5)
+	sum := fmt.Sprintf("%x", sha256.Sum256([]byte(endpointID)))
```
Comment (aanm) [inline pkg/endpoint/connector/add.go:43]: @nirmoy Won't this end up having the same problem? The first 5 characters of the sha256 can still colide. What about using the hexadecimal format of the endpoint ID without being truncated?

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/endpoint/connector/add.go (hunk 1 above)
Comment (nirmoy) [reply pkg/endpoint/connector/add.go:43]: @aanm len(namespace) can be > len(ifname). 
chapman:~ # ip netns add 1234567890123456783
chapman:~ # ip link add 123456789012346 type dummy 
chapman:~ # ip link add 1234567890123467 type dummy 
Error: argument "1234567890123467" is wrong: "dev" not a valid ifname

Comment (aanm) [reply pkg/endpoint/connector/add.go:43]: @nirmoy I thought the endpointID of the function was the endpoint ID and not the container ID. I was suggestion we could simply use the endpoint ID for the Ifname
```bash
"lxc${cilium_endpoint_id_in_hex}"
```

Comment (nirmoy) [reply pkg/endpoint/connector/add.go:43]: updated..

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
As always, awesome finding!

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Pull request merged**